design-proposals/data-import-cron-http-import.md

```diff
@@ -0,0 +1,78 @@
+The problem that arises from HTTP imports is that there is no convention between the different sources, so it's hard to know when the image is updated for each source in a generic way, which will make the polling process more difficult than standard registry sources.  
+One possible solution If the URL is static is to use a Get request with an If-Modified-Since header where we specify the date from which we want to check if there was a change. If there was a change since the specified date, the request will return with a status of 200OK, and then we know that the image has been updated since that date.  
+
+In the current situation we can enable HTTP imports by updating the DesiredDigest label.  
+
+## Goals
+
+* Allow the user to perform http imports manually with dataimportcron
+* Create a poller that will cover as many cases as possible for automatic updating
+
+## Non Goals
+
+* The poller will probably not cover all import cases and sometimes the user will have to manually update the digest
+
+## User Stories
+
+* As a user, I would like to import images from an HTTP source using DataImportCron
+* As a user, I would like the poller automatically update the image when it is needed
+* As a user, I would like to manually trigger an HTTP import with DataImportCron
+
+## Repos
+
+* **CDI**: Offers an API which serves as a mutable symbolic link to a Persistent Volume Claim (PVC). This link can be used in a DataVolume.
+* **CDI**: Provides an API called DataImportCron, designed to manage the periodic polling and importing of disk images as PVCs into a specific namespace (golden images by default). This functionality leverages the CDI import processes.
+
+## Implementation Phases
+
+**Phase1** - Manual Import Update
+* DataImportCron controller to handle HTTP imports
+* DataImportCron webhook/validation to support HTTP source type
+
+**Phase2** - Poller
+* DataImportCron poller to support HTTP source polling
+
+# Design
+
+If the URL is static:
+* Make a GET request with the If-Modified-Since Header starting from the date stored in the AnnLastCronTime annotation
+* If the returned status is 200OK, perform import again
+* Update AnnLastCronTime to time.Now()
+
+## DataImportCron Example
+
+This example polls http://tinycorelinux.net/14.x/x86/release/Core-14.0.iso static image. The DataImportCron will import it as new PVCs and automatically manage updating the corresponding DataSource.
+
+* **source** specifies where to poll from
+* **http** specifies that the type is HTTP import
+* **url** specifies the image URL
+* **schedule** should be empty when we are manually updating the DataImportCron desired digest annotation
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataImportCron
+metadata:
+  name: fedora-image-import-cron
+  namespace: golden-images
+spec:
+  template:
+    spec:
+      source:
+        http:
+          url: "http://tinycorelinux.net/14.x/x86/release/Core-14.0.iso"
```

Comment:
If the URL is static then it can help, if it changes so we need to find out the new suffix in addition to the hashcode to know the new URL. There could be other issues like the format of the checksum file.
This can help with Fedora case, for example `Fedora-Cloud-Base-38-1.6.x86_64.qcow2` supposed to stay the same.

Maybe we can use the name of the version that is in the same file you mentioned (e.g `CorePure64-14.0.iso`). The question is whether the name always changes when the hashcode changes or it is possible to push changes to an existing image even if it is not named differently.

Reply:
If for some sources the filename `Fedora-Cloud-Base-38-1.6.x86_64` changes things get more
tricky for us, we might need some API on the DV level that is similar to sourceRef which derives
the new filename from the checksum file

Reply:
Maybe some templating type thing could be useful. But I think we should start out with just considering 'static' URLs, get that working properly, then expand to more complex use cases if what we have is not sufficiently powerful.